src/main/java/org/kitteh/irc/client/library/feature/SimpleDefaultingOutboundMessageMap.java

```diff
@@ -0,0 +1,144 @@
+
+        this.defaultString = defaultString;
+    }
+
+    /**
+     * Creates a SimpleDefaultingOutboundMessageMap object with predefined string to default
+     * to when an DefaultingOutboundMessage key is not set.
+     */
+    public SimpleDefaultingOutboundMessageMap() {
+        this.defaultString = null;
+    }
+
+    /**
+     * Sets a default message for the DefaultingOutboundMessage.
+     *
+     * @param key The interaction to set a value for
+     * @param defaultString The default value to assume when nothing is
+     * provided for this DefaultingOutboundMessage
+     * @return Returns this to permit chaining
+     * @throws IllegalArgumentException Will throw an IllegalArgumentException is you set the
+     * defaultString value to "". Use null instead.
+     */
+    public SimpleDefaultingOutboundMessageMap setDefault(@Nonnull DefaultingOutboundMessage key, @Nullable String defaultString) {
+        Sanity.nullCheck(key, "DefaultingOutboundMessage key must not be null");
+        defaultString = this.emptyStringCheck(defaultString);
+
+        this.defaults.put(key, defaultString);
+
+        return this;
+    }
+
+    /**
+     * Retrieves the default string value for the specified
+     * DefaultingOutboundMessage.
+     *
+     * @param key the default DefaultingOutboundMessage key to obtain the
+     * default of
+     * @return the string of the default value, or the predefined
+     * defaultString
+     */
+    public Optional<String> getDefault(DefaultingOutboundMessage key) {
+        return this.getDefault(key, this.defaultString);
+    }
+
+    /**
+     * Retrieves the default string value for the specified
+     * DefaultingOutboundMessage, but override the defaultString
+     * value with our own message if the value is not set.
+     *
+     * @param key default DefaultingOutboundMessage key to get the default of
+     * @param defaultValue instead of referencing to the defaultString,
+     * use this value instead if it is not set.
+     * @return string of the default value, or the second parameter
+     * if the key is not set.
+     */
+    public Optional<String> getDefault(@Nonnull DefaultingOutboundMessage key, @Nullable String defaultValue) {
+        Sanity.nullCheck(key, "DefaultingOutboundMessage key must not be null");
+
+        String value = this.defaults.get(key);
+
+        return value != null ? Optional.of(value) : Optional.ofNullable(defaultValue);
+    }
+
+
+    /**
+     * Retrieves a list of all currently-defined defaults, filling in the
+     * unset keys with the defaultString defined in the object's constructor.
+     *
+     * @return a map with default strings for all interactions
+     */
+    public Map<DefaultingOutboundMessage, String> getDefaults() {
+        for (DefaultingOutboundMessage key : DefaultingOutboundMessage.values()) {
+            this.defaults.computeIfAbsent(key, value -> this.defaultString);
+        }
+
+        return this.defaults;
```

Comment:
Should this be mutable? I don't think so.
`return Collections.unmodifiableMap(this.defaults);` or a hard-copy.